core/src/main/java/io/questdb/cutlass/pgwire/PGConnectionContext.java

```diff
@@ -413,18 +415,34 @@ public void handleClientOperation(
                 // we do not pre-compute length because 'parse' will mutate 'recvBufferReadOffset'
                 if (keepReceiving) {
                     do {
-                        // Parse will update the value of recvBufferOffset upon completion of
+                        // Parse will update the value of recvBufferReadOffset upon completion of
                         // logical block. We cannot count on return value because 'parse' may try to
                         // respond to client and fail with exception. When it does fail we would have
                         // to retry 'send' but not parse the same input again
 
                         long readOffsetBeforeParse = recvBufferReadOffset;
                         totalReceived += (recvBufferWriteOffset - recvBufferReadOffset);
-                        parse(
-                                recvBuffer + recvBufferReadOffset,
-                                (int) (recvBufferWriteOffset - recvBufferReadOffset),
-                                compiler
-                        );
+                        try {
+                            parseMessage(
+                                    recvBuffer + recvBufferReadOffset,
+                                    (int) (recvBufferWriteOffset - recvBufferReadOffset),
+                                    compiler
+                            );
+                        } catch (SqlException e) {
```

Comment:
can we unnest exception handing pls?

Reply:
yeah, I was unsure how exactly the resume processors work. let me have a look, it should be possible to simplify it.